### ISSUE TITLE
AWS Elasticache Redis Cluster : Uncaught PHP Exception Predis\Response\ServerException: MOVED

### ISSUE THREAD

**Issue by ebarault** [snc/SncRedisBundle#402]
Hi, 

AWS provides a single configuration endpoint for an Elasticache Redis Cluster (x shards, y replicas per shards).

With the following configuration, i get this error: 
```
Uncaught PHP Exception Predis\Response\ServerException: "MOVED 2777 w.x.y.z:6379" at /var/www/html/vendor/predis/predis/src/Client.php line 370
```

```
snc_redis:
    clients:
       default:
         type: predis
         alias: default
         dsn:
           - my-redis.cmfj2h.ng.0001.euw3.cache.amazonaws.com:6379
         options:
           profile: 3.2
           cluster: redis
           replication: false
```

I use snc_redis bundle **2.0.4** and predis bundle **1.1.1**

I find very little doc (mostly in issues) about server-side replication/sharding support option (`cluster: redis`).

Has anyone solved this?

**Comment by curry684** [comment]
Given the exotic setup I'm not sure we're going to be able to give this issue much love soon. Are you by any chance able to debug this yourself and possibly work towards a PR?

**Comment by ebarault** [comment]
I could try to work this out, but in what sense do you qualify the setup exotic? Up to my understanding AWS Elasticache's configuration endpoint is just AWS's name for the single point of contact for a server-side managed redis cluster.

**Comment by ebarault** [comment]
to help me getting started with the debug: how can i verify that `options.cluster = redis` actually sets the right cluster mode?
from various comments, `MOVED` errors indicate that the client does not use the server-managed cluster mode

**Comment by curry684** [comment]
I would love to help, but right now I'm afraid you're pretty much on your own. The bundle hasn't had much maintenance over the past year, and I'm stepping in now with @Seldaek to do some cleanup and all, but specifically on the clustering features I don't think either of us are really knowledgeable right now until we've had time to dive into specifics a bit more.

**Comment by curry684** [comment]
@ebarault are you in a position to test whether #416 fixes your issues?

**Comment by ebarault** [comment]
@curry684 : yes, i'll try this later today or tomorrow

**Comment by ebarault** [comment]
@curry684 : did a first try
with `replication: true`
got this exception
```
No master server available for replication
500 Internal Server Error - MissingMasterException
```
(full trace available)

I'm not sure if this option is meant to be used with cluster mode `redis` though. Any idea?

with `replication: false`, for some reason it currently breaks one of our composer scripts (`assets:install`), and couldn't push it any further at the moment. We will try to debug this one on the coming days.

**Comment by curry684** [comment]
Let's discuss the possible solution in #416   :)

**Comment by curry684** [comment]
Fixed by #416
